modules/ossm-creating-an-egress-gateway-for-http-traffic.adoc

```diff
@@ -0,0 +1,191 @@
++
+[source,terminal]
+----
+$ oc create namespace test-namespace
+----
+
+. Enable sidecar injection in the namespace:
++
+[source,terminal]
+----
+$ oc label namespace test-namespace istio-injection=enabled
+----
+
+Deploy the sleep application in the namespace:
++
+[source,terminal]
+----
+$ oc apply -f https://raw.githubusercontent.com/istio/istio/release-1.23/samples/sleep/sleep.yaml -n test-namespace
+----
+
+. Display information about the pods in the namespace:
++
+[source,terminal]
+----
+$ oc get pods -n test-namespace
+----
+
+. Store the sleep pod in the `SOURCE_POD` environment variable:
++
+[source,terminal]
+----
+$ export SOURCE_POD=$(oc get pod -n test-namespace -l app=sleep -o jsonpath='{.items[0].metadata.name}')
+----
+
+. Create a `ServiceEntry` to allow direct traffic from the mesh to an external service. The following example defines a `ServiceEntry` for a URL.
++
+.Sample configuration
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: networking.istio.io/v1
+kind: ServiceEntry
+metadata:
+  name: <example_egress>
+  namespace: test-namespace
+spec:
+  hosts:
+    - <example_url.com>
+  ports:
+    - number: 80
+      name: http-port
+      protocol: HTTP
+    - number: 443
+      name: https
+      protocol: HTTPS
+  location: MESH_EXTERNAL
+  resolution: DNS
+----
+
+. Send an HTTP request to the URL that you used in the previous step to verify that the `ServiceEntry` was applied correctly.
++
+[source,terminal]
+----
+$ oc exec "$SOURCE_POD" -n test-namespace -c sleep -- curl -sSL -o /dev/null -D - <example_url>
+----
++
+The terminal should display information similar to the following:
+
+.Example output
+[source,terminal]
+----
+...
+HTTP/1.1 301 Moved Permanently
+...
+location: <example_url>
+...
+
+HTTP/2 200
+Content-Type: text/html; charset=utf-8
+----
+
+. Create an egress `Gateway` that routes traffic from the mesh to the port that you selected for the external service.
++
+.Sample configuration
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: <example_egress_gateway>
+  namespace: istio-gateway
+spec:
+  selector:
+    istio: egress-gateway
+  servers:
+    - port:
+        number: 80
+        name: http
+        protocol: HTTP
+      hosts:
+        - <example_url>
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: egressgateway-for-<example_url>
+  namespace: istio-gateway
+spec:
+  host: egress-gateway.istio-gateway.svc.cluster.local
+  subsets:
+    - name: <name>
+----
+
+. Configure the routing rules that direct traffic from the application sidecar proxy to the egress gateway and from the egress gateway to the external service.
++
+.Sample configuration
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: egress-gateway-vs
+  namespace: test-namespace
```

Comment:
namespace: test-namespace # Namespace where the `sleep` pod is running

Reply:
done